### PR TITLE
Results-journal recommendation web app

### PR DESCRIPTION
**Pull request by danich1**
This PR is the last results section of this manuscript. This section talks about the journal recommendation web app and the machine learning model performance.

### CODE REVIEW

File: content/04.results.md [hunk 1]
```diff
@@ -169,6 +170,26 @@ Our filtering criteria were intentionally stringent, so the increased estimate o
...
 As opposed to those published in 2019 and later, these preprints are old enough that they are likely to have been published but it was interesting that the rate was not observed to be higher for older preprints.
 
-### Journal Recommendations/Audience Associations
-1. Title will change once analysis is finished 
-2. Provide key figure for this section and take-home message
+### Recommending Journals Based on Preprint Representation
+
+![
+Both classifiers outperform the randomized baseline when predicting a paper's journal endpoint.
+This bargraph shows each model's accuracy in respect to predicting the training and test set.
```
Comment (cgreene) [inline content/04.results.md:177]: How is the training set accuracy in this classifier not 100%? I thought our evaluation was whether or not the eventually published journal was in the top X (where X is 10 in this case b/c of the webserver).

File: content/04.results.md [hunk 2]
```diff
@@ -169,6 +170,26 @@ Our filtering criteria were intentionally stringent, so the increased estimate o
...
+![
+Here is the workflow of the journal recommender web-app.
+Starting with the homescreen users can paste in a _bioRxiv_ or _medRxiv_ doi, which sends a request to biorxiv or medrxiv (A).
+Next our app preprocesses the preprint and returns a listing of the top ten most similar papers (B) and the top ten closest journals to the query (C).
+Following the listing, our app manually plots the preprint query onto the Pubmed Central Landscape (D).
+Lastly, users can click on a square within the landscape, which will show bin statistics as well as associated word-odd ratios (E).
+](images/journal_recommender_workflow.png){@fig:journal_rec_workflow}
+
+We constructed an online app that provides users with journal suggestions based on their preprint content.
+This tool takes in doi links from _bioRxiv_ and _medRxiv_ and provides the user with the following output: title of the query preprint, the top ten most similar papers, the top ten journal recommendations, the location where the query preprint falls along the PMC landscape and statistics on individual bins within the landscape (Figure {#fig:journal_rec_workflow}).
```
Comment (cgreene) [inline content/04.results.md:195]:
```suggestion
Users supply DOIs from _bioRxiv_ or _medRxiv_.
The application then downloads the article, converts the PDF to text, calculates a document embedding score, and returns the ten papers and journals with the most similar representations in the embedding space.
It also embeds the document into the overall PMC landscape for visualization and allows the user to examine principal components and term enrichment for each bin within the landscape (Figure {#fig:journal_rec_workflow}).
```

**Review by cgreene — COMMENTED** [review]
One question and then some suggestions on the app.

File: content/04.results.md (hunk 1 above)
Comment (danich1) [reply content/04.results.md:177]: >How is the training set accuracy in this classifier not 100%? I thought our evaluation was whether or not the eventually published journal was in the top X (where X is 10 in this case b/c of the webserver).

You are correct in that the evaluation is determining if the published journal appears within the top 10 nearest neighbors. The problem here is that a paper could align closely to other papers from drastically different journal endpoints rather than a paper from the true journal endpoint. These "other" papers drown out the desired signal, which causes a decrease in performance for the paper-paper classifier. Regarding the journal centroid classifier, it could be the similar issue as above. Outside journals could be publishing more papers that align better to the query rather than the true journal itself. 

The numbers in this chart are for the subsampled training set version (randomly selecting 100 random papers from each journal); however, even when using the full dataset the max performance was 62% (mega-journal influence).

Comment (cgreene) [reply content/04.results.md:177]: Is a paper not necessarily most similar to itself though? That's the bit that I'm confused by.

Comment (danich1) [reply content/04.results.md:177]: >Is a paper not necessarily most similar to itself though? That's the bit that I'm confused by.

A paper should be most similar to itself, but might not be most similar to _other_ papers from the same journal.

Comment (cgreene) [reply content/04.results.md:177]: If it's most similar to itself, wouldn't its journal always be in the top 10 though?

Comment (danich1) [reply content/04.results.md:177]: As discussed in person. The confusion here is the figure says evaluated on training set. This is true; however the training set was evaluated via _cross validation_. Will need to update the figure to say cross validation instead of training set to avoid any further confusion.

File: content/04.results.md [hunk 3]
```diff
@@ -169,6 +170,28 @@ Our filtering criteria were intentionally stringent, so the increased estimate o
...
 As opposed to those published in 2019 and later, these preprints are old enough that they are likely to have been published but it was interesting that the rate was not observed to be higher for older preprints.
 
-### Journal Recommendations/Audience Associations
-1. Title will change once analysis is finished 
-2. Provide key figure for this section and take-home message
+### Recommending Journals Based on Preprint Representation
+
+![
+Both classifiers outperform the randomized baseline when predicting a paper's journal endpoint.
+This bargraph shows each model's accuracy in respect to predicting the training and test set.
```
Comment (cgreene) [inline content/04.results.md:177]:
```suggestion
This bargraph shows each model's accuracy as assessed through cross validation or an independent held out test set.
```

File: content/04.results.md [hunk 4]
```diff
@@ -169,6 +170,28 @@ Our filtering criteria were intentionally stringent, so the increased estimate o
...
+](https://raw.githubusercontent.com/danich1/annorxiver/6bedc581a88b8d98043d847d234d06877458d898/pmc/journal_recommendation/output/figures/knn_result.png){@fig:knn_auc}
+
+We sought to recommend journals based on the textual content of a paper.
+We trained two different classifiers (centroid classifier and paper-paper classifier) to predict the journal endpoints for already published papers.
```
Comment (cgreene) [inline content/04.results.md:181]:
```suggestion
We trained two different classifiers to predict the journal endpoints for already published papers. 
One classifier uses the nearest journal centroids, which attempts to capture the topic area of a journal.
The other classifier aims to be more granular and uses the journals that published the nearest papers.
```

File: content/04.results.md [hunk 5]
```diff
@@ -169,6 +170,28 @@ Our filtering criteria were intentionally stringent, so the increased estimate o
...
-1. Title will change once analysis is finished 
-2. Provide key figure for this section and take-home message
+### Recommending Journals Based on Preprint Representation
+
+![
+Both classifiers outperform the randomized baseline when predicting a paper's journal endpoint.
+This bargraph shows each model's accuracy in respect to predicting the training and test set.
+](https://raw.githubusercontent.com/danich1/annorxiver/6bedc581a88b8d98043d847d234d06877458d898/pmc/journal_recommendation/output/figures/knn_result.png){@fig:knn_auc}
+
+We sought to recommend journals based on the textual content of a paper.
```
Comment (cgreene) [inline content/04.results.md:180]:
```suggestion
We sought to identify journals that might publish a preprint based on the text of a paper.
```

File: content/04.results.md [hunk 6]
```diff
@@ -169,6 +170,28 @@ Our filtering criteria were intentionally stringent, so the increased estimate o
...
+We trained two different classifiers (centroid classifier and paper-paper classifier) to predict the journal endpoints for already published papers. 
+We found that both clasifiers are better than our randomized baseline and that our centroid classifier acheived better accuracy on the test set compared to the paper-paper classifier (Figure {#fig:knn_auc}).
```
Comment (cgreene) [inline content/04.results.md:182]:
```suggestion
Both classifiers outperformed a randomized baseline.
A classifier that aimed to predict centroids performed better on the held out test set compared than the nearest paper classifier (Figure {#fig:knn_auc}).
```

File: content/04.results.md [hunk 7]
```diff
@@ -169,6 +170,28 @@ Our filtering criteria were intentionally stringent, so the increased estimate o
...
+Despite the positive performance, recommending journals remains a challenging task as both model's overall accuracy is modest at best (Figure {#fig:knn_auc}).
```
Comment (cgreene) [inline content/04.results.md:183]:
```suggestion
There are [HOW MANY] journals in our dataset, so the baseline performance of a classifier is quite low.
We were able to achieve a substantial increase with respect to random performance at predicting the journals that a paper was published in.
However, the predictor is not perfect (Figure {#fig:knn_auc}), which we should expect because there are multiple journals that cover certain topic areas and others have a very broad set of covered topic areas.
Still, our software provides a starting point for authors to use the text of their preprints to identify potentially suitable publication venues.
```

**Review by cgreene — APPROVED** [review]
Makes sense now! One more set of suggestions & can you change the figure label before merging? 🚀
